cqrs/src/main/scala/it/pagopa/interop/commons/cqrs/service/ReadModelService.scala

```diff
@@ -72,6 +72,16 @@ final class MongoDbReadModelService(dbConfig: ReadModelConfig) extends ReadModel
       model   <- results.traverse(extractData[T](_).toFuture)
     } yield model
 
+  def aggregateRaw[T: JsonReader](collectionName: String, pipeline: Seq[Bson], offset: Int, limit: Int)(implicit
+    ec: ExecutionContext
+  ): Future[Seq[T]] = for {
+    results <- db
+      .getCollection(collectionName)
+      .aggregate(pipeline ++ Seq(Aggregates.skip(offset), Aggregates.limit(limit)))
+      .toFuture()
+    model   <- Future.traverse(results)(extract[T](_).toFuture)
+  } yield model
```

Comment:
just to avoid code duplication, even if minor (not sure it compiles, but just to give an idea)
```suggestion
  def aggregateRaw[T: JsonReader](collectionName: String, pipeline: Seq[Bson], offset: Int, limit: Int)(implicit
    ec: ExecutionContext
  ): Future[Seq[T]] = aggregator(collectionName, pipeline, offset, limit)(extract)
  
  def aggregator[T: JsonReader](collectionName: String, pipeline: Seq[Bson], offset: Int, limit: Int)(extractor: document: Document => Either[Throwable, T])(implicit
    ec: ExecutionContext
  ): Future[Seq[T]] = for {
    results <- db
      .getCollection(collectionName)
      .aggregate(pipeline ++ Seq(Aggregates.skip(offset), Aggregates.limit(limit)))
      .toFuture()
    model   <- Future.traverse(results)(extractor[T](_).toFuture)
  } yield model
```
Changing accordingly also the `aggregate` function

Reply:
The idea is to use it to refactor the other method as well, right?

Reply:
exactly

Reply:
I pushed an implementation, can you take a look? :D